tools/serverpod_cli/lib/src/analyzer/entities/entity_parser/entity_parser.dart

```diff
@@ -168,42 +168,53 @@ class EntityParser {
     );
 
     var scope = _parseClassFieldScope(value);
-    var parentTable = _parseParentTable(value);
     var shouldPersist = _parseShouldPersist(value);
-    var scalarFieldName = _parseScalarField(value, fieldName);
     var isEnum = _parseIsEnumField(value);
 
-    RelationDefinition? relation;
+    var parentTable = _parseParentTable(value);
+    var scalarFieldName = _parseScalarField(value, fieldName);
+    var manualScalarField = _parseManualScalarField(value);
```

Comment:
Question: Couldn't `_parseManualScalarField` just have the default value from `_parseScalarField` then it would just be one call? Why do we have the comparison and the logic on the outside?

Reply:
Because if we have a manual field there should be no automatic scalar field and vise versa. If there is a manual one we should not inject an extra field into our class that holds the relation, but if it is automatic we should inject it. 

In other words there is a different flow and outcome if the manual or automatic is set.